### DISCUSSION TITLE
V3 seems to rerender whole app on search params change

### DISCUSSION THREAD

**Discussion by rlesniak** [TanStack/router#571] [General]
We got a pretty big app, one of our view has complex logic under one of routes. After testing, we noticed that adding parameter to search params causes to rerender whole app, which takes ~800ms.
In below screenshot we dont even utilize param `stuff` anywhere in the app but it still gets to be rerendered.

Example code
```
onClick={() => {
           updateSearch({
            stuff: 'YO',
          });
         // setSelectedPlanItem(item);
}}
```
![image](https://user-images.githubusercontent.com/13970783/229143760-66b013f0-728e-4b4d-a072-bee3209af124.png)

and example with same onClick but with changing something using zustand

```
onClick={() => {
          // updateSearch({
          //   stuff: 'YO',
          // });
          setSelectedPlanItem(item);
}}
```

![image](https://user-images.githubusercontent.com/13970783/229144117-f18c663f-3c37-4f3c-acab-f1eecca09abd.png)

### Conclusion
So I think what react-location lacks is the selectors that react-redux or zustand have, which allows to skip re-rendering if the selected slice is the same.

Am I right or I am missing something?

Tested on version `3.7.4`

**Comment by zlanich** [comment]
I'm seeing this too. I'm using Nuqs to put datatable query params in the URL, so every time I change my datatable search, the whole app re-renders, even if it is on a debounce. This doesn't seem right to me. Seems like only things that subscribe to route changes to re-render. What am I missing?
